src/cli.ts

```diff
@@ -55,38 +91,127 @@ function showHelp() {
+  console.log();
+  console.log('  Generates constructs in Python under the "src" subfolder instead of working directory:');
+  console.log('    cdk-import sc -l python -o ./src');
+  console.log();
+  console.log();
+}
+
 void (async () => {
-  if (args.help || args._.length !== 1) {
+  if (args.help) {
     showHelp();
     process.exit(1);
   }
 
   if (!args.language) {
-    throw new Error(`Missing required option: --language. Supported languages: ${SUPPORTED_LANGUAGES.join(',')}`);
+    throw new Error(`Missing required option: --language. Supported languages: ${SUPPORTED_LANGUAGES.join(', ')}`);
   }
 
   if (SUPPORTED_LANGUAGES.indexOf(args.language) === -1) {
-    throw new Error(`Unsupported language ${args.language}. Supported: ${SUPPORTED_LANGUAGES.join(',')}`);
+    throw new Error(`Unsupported language ${args.language}. Supported: ${SUPPORTED_LANGUAGES.join(', ')}`);
   }
 
-  try {
-    const [resourceName, resourceVersion] = args._[0].split('@');
-    const workdir = await fs.mkdtemp(path.join(os.tmpdir(), 'cdk-import'));
-    const typeName = await importResourceType(resourceName, resourceVersion, {
-      outdir: workdir,
-      private: args.private,
-    });
-
-    await renderCode({
-      srcdir: workdir,
-      language: args.language,
-      outdir: args.outdir ?? '.',
-      typeName: typeName,
-      goModule: args['go-module'],
-      javaPackage: args['java-package'],
-    });
-  } catch (e) {
-    console.log(e);
-    process.exit(1);
+  if (subCommand == 'cfn') {
+    if (args._.length !== 1) {
+      showCfnHelp();
+      process.exit(1);
+    }
+
+    try {
+      const [resourceName, resourceVersion] = args._[0].split('@');
+      const workdir = await fs.mkdtemp(path.join(os.tmpdir(), 'cdk-import'));
+      const typeName = await importResourceType(resourceName, resourceVersion, {
+        outdir: workdir,
+        private: args.private,
+      });
+
+      await renderCode({
+        srcdir: workdir,
+        language: args.language,
+        outdir: args.outdir ?? '.',
+        typeName: typeName,
+        goModule: args['go-module'],
+        javaPackage: args['java-package'],
+        csharpNamespace: args['csharp-namespace'] ?? typeName, // already AWS::Foo::Bar
+      });
+    } catch (e) {
+      console.log(e);
+      process.exit(1);
+    }
+  }
+
+  if (subCommand == 'sc') {
+    try {
+      const workdir = await fs.mkdtemp(path.join(os.tmpdir(), 'cdk-import'));
+      const ppArgs = [args['product-id'], args['provisioning-artifact-id'], args['path-id']].filter(param => param != undefined);
+
+      if (args.help || (ppArgs.length > 0 && ppArgs.length != 3)) {
+        showSCHelp();
+        process.exit(1);
+      }
+
+      if (!args['csharp-namespace'] && args.language == 'csharp') {
+        throw new Error('C# namespace name (`--csharp-namespace`) must be specified (e.g. "AWS::Foo::Bar")');
```

Comment:
is `AWS::Foo::Bar` an actual namespace name?

Reply:
thats more to show the format, I was referencing the comment from the cfn registry resource typename

Reply:
okay i see, will fix

Reply:
whoops great catch!  Pulled this from the cfn part, but yeah, we're not deriving from  a type here so the namespace should be a valid one.

Reply:
I don't even think an example is necessary.

Reply:
ended up removing this and defaulting to the product name, similar to the cfn case

Reply:
thus making csharp namespace optional for both subcommands